### ISSUE TITLE
User home systemProp properties not reflected when using the tooling API

### ISSUE THREAD

**Issue by DanielThomas** [gradle/gradle#17708]
System properties set in `gradle.properties` aren't reflected when using the tooling API.

### Expected Behavior
`systemProp` properties in `GRADLE_USER_HOME/gradle.properties` should be reflected when using the tooling API.

### Current Behavior
They are not, meaning it's not possible to configure system properties globally.

### Context
We have conditional logic in an init script `initscript` block that we need to allow an override for. Because properties aren't available there yet, we fallback to `startParameter` to get explicit project properties from the command line and the same key for the system property to allow it to be set via `gradle.properties`.

Ideally all properties would be available here, as they are for settings via the extra properties extension to avoid these fallbacks, but at least the `systemProp` configuration should be honored.

That works for command line builds, however if you set a `systemProp` property in the user home and refresh the project in IntelliJ for instance, the system property isn't set.

### Steps to Reproduce 
1. Add an init script to `GRADLE_USER_HOME/init.d`:
```
if (System.getProperty("myprop") == null) {
    throw new IllegalArgumentException("Property not set")
}
```
2. Add `systemProp.myprop=foo` to `GRADLE_USER_HOME/gradle.properties`
3. Invoke the build from the command line and note the property isn't set
4. Evaluate the project model using the tooling API (not verified, but assume due to what we're seeing from IntelliJ)

### Your Environment
Gradle 6.9, macOS

**Comment by Fiouz** [comment]
Works fine for me in Gradle 7.1, which addressed #6825.

**Comment by DanielThomas** [comment]
Oh that's hilarious: Martin is on my team. Same issue, I must have missed this in the release notes.
